test/functional/p2p_blocksonly.py

```diff
@@ -55,8 +55,28 @@ def run_test(self):
         with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=1'.format(txid)]):
             self.nodes[0].sendrawtransaction(sigtx)
             self.nodes[0].p2p.wait_for_tx(txid)
+            pprint(self.nodes[0].p2p.wait_for_tx(txid))
             assert_equal(self.nodes[0].getmempoolinfo()['size'], 1)
 
+        self.log.info("Restarting node 0 with whitelist permission and blocksonly")
+        self.restart_node(0, ["persistmempool=0", "-whitelist=127.0.0.1", "-whitelistforcerelay", "-blocksonly"])
```

Comment:
@MarcoFalke I think you meant "persistmempool=0"? I added it here, but I am also getting an error that says "[node 0] Error: no RPC connection" as a result. Adding the arg  "persistmempool=0" leads to the RPC error.